src/container.cpp

```diff
@@ -136,7 +134,7 @@ bool Container::unserializeItemNode(OTB::Loader& loader, const OTB::Node& node,
 			return false;
 		}
 
-		Item* item = Item::CreateItem(itemPropStream);
+		std::shared_ptr<Item> item = Item::CreateItem(itemPropStream);
```

Comment:
Could this be `auto` or `const auto`
```suggestion
		auto item = Item::CreateItem(itemPropStream);
```

What effect would using const auto here have? It seems that no subsequent code modifies this pointer, only the object’s data.
```suggestion
		const auto item = Item::CreateItem(itemPropStream);
```